ansible-tests/validations/library/validate_network_environment.py

```diff
@@ -0,0 +1,167 @@
+
+    # Look though every resources bridges and make sure there is only a single
+    # bond per bridge and only 1 interface per bridge if there are no bonds.
+    for name, resource in nic_data.get('resources', {}).items():
+        bridges = resource['properties']['config']['os_net_config']['network_config']
+        for bridge in bridges:
+            if bridge['type'] == 'ovs_bridge':
+                bond_count = 0
+                interface_count = 0
+                for bond in bridge['members']:
+                    if bond['type'] == 'ovs_bond':
+                        bond_count += 1
+                    if bond['type'] == 'interface':
+                        interface_count += 1
+                if bond_count == 2:
+                    errors.append(
+                        'Invalid bonding: There are 2 bonds for '
+                        ' bridge %s of resource %s in %s' %
+                        (bridge['name'], name, path))
+                if bond_count == 0 and interface_count > 1:
+                    errors.append(
+                        'Invalid interface: When not using a bond, '
+                        'there can only be 1 interface for bridge %s '
+                        'of resource %s in %s' %
+                        (bridge['name'], name, path))
+    return errors
+
+
+def check_cidr_overlap(networks):
+    errors = []
+    objs = []
+    for x in networks:
+        try:
+            objs += [netaddr.IPNetwork(x.decode('utf-8'))]
+        except ValueError:
+            errors.append('Invalid address: %s' % x)
+
+    for net1, net2 in itertools.combinations(objs, 2):
+        if (net1 in net2 or net2 in net1):
+            errors.append(
+                'Overlapping networks detected %s %s' % (net1, net2))
+    return errors
+
+
+def check_allocation_pools_pairing(filedata, pools):
+    errors = []
+    for poolitem in pools:
+        pooldata = filedata[poolitem]
+        pool_objs = [netaddr.IPRange(
+            netaddr.IPAddress(x['start'].decode('utf-8')),
+            netaddr.IPAddress(x['end'].decode('utf-8'))) for x in
+            pooldata]
+
+        subnet_item = poolitem.split('AllocationPools')[0] + 'NetCidr'
+        try:
+            subnet_obj = netaddr.IPNetwork(
+                filedata[subnet_item].decode('utf-8'))
+        except ValueError:
+            errors.append('Invalid address: %s' % subnet_item)
+
+        for ranges in pool_objs:
+            for range in ranges:
+                if range not in subnet_obj:
+                    errors.append('Allocation pool {} {} outside of subnet'
+                                  '{}: {}'.format(poolitem,
+                                                  pooldata,
+                                                  subnet_item,
+                                                  subnet_obj))
+                    break
+    return errors
+
+
+def check_vlan_ids(vlans):
+    errors = []
+    invertdict = {}
+    for k, v in vlans.iteritems():
```

Comment:
I'm afraid you'll have to resort to six to make your code compatible with python 2 and 3.